src/UnderlineNav2/UnderlineNavItem.tsx

```diff
@@ -179,7 +135,7 @@ export const UnderlineNavItem = forwardRef(
               as="span"
               data-component="text"
               data-content={children}
-              sx={selectedLink === ref ? {fontWeight: 600} : {}}
+              sx={Boolean(ariaCurrent) && ariaCurrent !== 'false' ? {fontWeight: 600} : {}}
```

Comment:
@broccolinisoup apologies if this PR isn't related to design feedback, but I noticed that this version handles the selected "bold" state differently than the Rails counterpart. When we change the font to a different weight, you'll notice the item width expand since the text is literally taking up more space. In Rails, we're using a "fake" kind of bold where we repeat the item label using CSS content. I'm not sure if it used to be that way here and was removed or if it just hasn't been addressed yet but wanted to bring it up!

Reply:
@langermank We still have [the fake bold](https://github.com/primer/react/blob/57e71d847d6790f7894a54b8c510595d3e736dec/src/UnderlineNav2/styles.ts#L110) to make space for the bold selected item.

Here we set the `font weight` to actually make it bold as it is selected and the other items don't shift because we have the space as well.

**Currently**

https://github.com/primer/react/assets/1446503/b0717f1f-f6f3-432a-b0f8-f07f63001c98

**When I remove the fake bold ::before class**

https://github.com/primer/react/assets/1446503/6b93c8e8-49df-46c2-b2f5-e181d676a2c7

Let me know if I am missing anything??